snippets/csharp/VS_Snippets_CLR_System/system.collections.generic.equalitycomparer.default/cs/program.cs

```diff
@@ -1,128 +1,96 @@
-﻿// <Snippet1>
-using System;
+﻿using System;
 using System.Collections.Generic;
 
-class Program
+static class Program
 {
-    static void Main(string[] args)
+    static void Main()
     {
-        BoxEqDimensions bxd = new BoxEqDimensions();
-        BoxEqVolume bxv = new BoxEqVolume();
+        Box redBox = new Box(8, 8, 4);
+        Box blueBox = new Box(6, 8, 4);
+        Box greenBox = new Box(4, 8, 8);
 
-        Dictionary<Box, string> boxesByDim = new Dictionary<Box, string>(bxd);
-        Dictionary<Box, string> boxesByVol = new Dictionary<Box, string>(bxv);
+        var boxes = new CustomCollection<Box>(new[] { redBox, blueBox, greenBox });
 
-        try
-        {
-            Box redBox = new Box(8, 8, 4);
-            Box blueBox = new Box(6, 8, 4);
-            Box greenBox = new Box(4, 8, 8);
+        var foundByDimension = boxes.FindFirst(greenBox);
 
-            Console.WriteLine("Adding boxes by Dimension");
+        Console.WriteLine($"Found box {foundByDimension} by dimension.");
 
-            boxesByDim.Add(redBox, "red");
-            boxesByDim.Add(blueBox, "blue");
-            boxesByDim.Add(greenBox, "green");
+        var foundByVolume = boxes.FindFirst(greenBox, new BoxEqVolume());
 
-            PrintBoxCollection(boxesByDim);
+        Console.WriteLine($"Found box {foundByVolume} by volume.");
+    }
+}
 
-            Console.WriteLine("\nAdding boxes by Volume");
+public class CustomCollection<T>
+{
+    private IEnumerable<T> items;
 
-            boxesByVol.Add(redBox, "red");
-            boxesByVol.Add(blueBox, "blue");
-            boxesByVol.Add(greenBox, "green");
+    public CustomCollection(IEnumerable<T> items) => this.items = items;
 
-            PrintBoxCollection(boxesByVol);
-        }
-        catch (ArgumentException argEx)
-        {
-            Console.WriteLine(argEx.Message);
-        }
-    }
-    private static void PrintBoxCollection(Dictionary<Box,string> boxes)
+    public T FindFirst(T itemToFind, IEqualityComparer<T> comparer = null)
```

Comment:
To remove ceremony with defining and creating `CustomCollection<T>` and make the example further shorter and simpler, what about creating an extension method for the `List<T>` class that would extend [`IndexOf(T)`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.indexof?view=netcore-2.0#System_Collections_Generic_List_1_IndexOf__0_) method:
```csharp
public int FirstIndexOf(this List<T> list, T item, IEqualityComparer<T> comparer = null)
```
I suggest to call it `FirstIndexOf` to avoid calling the framework method when no `comparer` argument is given.

Reply:
Making it an extension method is a good idea. Though I haven't changed it to `FirstIndexOf`, I think returning the item is simpler and easier to understand.